pkg/yurthub/locallb/iptables.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package locallb
+
+import (
+	"strconv"
+
+	"k8s.io/klog/v2"
+
+	"github.com/coreos/go-iptables/iptables"
+)
+
+type IptablesManager struct {
+	ipt iptables.IPTables
+}
+
+func NewIptablesManager() *IptablesManager {
+	iptable, _ := iptables.New()
+	m := &IptablesManager{
+		ipt: *iptable,
+	}
+	return m
+}
+
+func (im *IptablesManager) updateIptablesRules(tenantKasService string, ips []string) error {
+	if err := im.cleanIptablesRules(tenantKasService, ips); err != nil {
+		return err
+	}
+	if err := im.addIptablesRules(tenantKasService, ips); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (im *IptablesManager) addIptablesRules(tenantKasService string, ips []string) error {
+	ramdomBalancingProbability := im.getRamdomBalancingProbability(len(ips))
+	for index, ip := range ips {
+		// All packets (from kubelet, kubeproxy, pods, etc.) are loadbalanced to multiple addresses of apiservers deployed in daemonset, except packets which are sent to host control plane.
+		err := im.ipt.Append("nat", "OUTPUT", "! -d", tenantKasService, "-p", "tcp", "-m", "statistic", "--mode", "random", "--probability", strconv.FormatFloat(ramdomBalancingProbability[index], 'f', -1, 64), "-j", "DNAT", "--to-destination", ip)
```

Comment:
we need to add port for the iptables rule, because source port and dest port maybe different.